OpenRA.Mods.Common/Traits/Attack/AttackTurreted.cs

```diff
@@ -17,17 +17,22 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Actor has a visual turret used to attack.")]
 	public class AttackTurretedInfo : AttackFollowInfo, Requires<TurretedInfo>
 	{
+		[Desc("Turret names")]
+		public readonly string[] Turrets = { "primary" };
+
 		public override object Create(ActorInitializer init) { return new AttackTurreted(init.Self, this); }
 	}
 
 	public class AttackTurreted : AttackFollow
 	{
+		public readonly AttackTurretedInfo AttackTurretedInfo;
```

Comment:
This doesn't look very nice.  Just cast it when you need it instead?

Reply:
Not sure what you mean with casting?
I couldn't come up with anything else that compiled and allowed looking up this AttackTurreted-specific Info in Turreted.